src/main.cpp

```diff
@@ -21,11 +21,20 @@ Copyright (C) 2011, Parsian Robotic Center (eew.aut.ac.ir/~parsian/grsim)
 
 int main(int argc, char *argv[])
 {
+    char** argend = argc + argv;
+
     QCoreApplication::setOrganizationName("Parsian");
     QCoreApplication::setOrganizationDomain("parsian-robotics.com");
     QCoreApplication::setApplicationName("grSim");
     QApplication a(argc, argv);
     MainWindow w;
-    w.show();
+
+    if (std::find(argv, argend, std::string("--headless")) == argend
+        && std::find(argv, argend, std::string("-H")) == argend) {
```

Comment:
I think there's typo here, it should be `||`

Reply:
This is actually correct, as it is 

if "--headless" is not found && if "-H" is not found, don't enable headless mode.

However, this logic is backwards and confusing (as pointed out), so I'll reverse the logic.

Reply:
yes, you're right.
Thanks, btw.